### PR TITLE
Updating the open-with-app to use the new getSelectedFinderItems command

### PR DESCRIPTION
**Pull request by omarshahine**
## Description

Updating the open-with-app to use the new [getSelectedFinderItems](https://developers.raycast.com/api-reference/environment) command

## Screencast

n/a

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by fturcheti** [comment]
Thanks for doing this long overdue update, @omarshahine.

**Comment by pernielsentikaer** [comment]
Could you both try it out 😊

**Comment by fturcheti** [comment]
@pernielsentikaer With these changes I'm not able to open multiple files at once. Previously, for example, I could select multiple images, invoke the command, and open all of them with a specific application. After the proposed changes, when I do so, only one image gets opened.

**Comment by fturcheti** [comment]
@pernielsentikaer Actually, even the old implementation is not working correctly. It can open multiple files, but it is not reliable. If I have, for example, 5 files selected, it is opening only 2 or 3.

**Comment by omarshahine** [comment]
> Actually, even the old implementation is not working correctly. It can open multiple files, but it is not reliable. If I have, for example, 5 files selected, it is opening only 2 or 3.
> 
> Although I don't use the extension to open multiple files quite often, I think it used to work properly.
> 
> @pernielsentikaer Do you know how we can make this work for multiple files?

Odd, so this close block does not work deterministically?

`onAction={async () => {
              try {
                const selectedItems = await getSelectedFinderItems();
                if (selectedItems.length != 0) {
                  selectedItems.forEach((item) => {
                    open(item.path.replace(/"/g, '\\"'), application.bundleId);
                  });
                } else {
                  await showHUD("⚠️  No Finder selection to open.");
                }
              } catch (error) {
                await showHUD("⚠️  No Finder selection to open.");
              }
              closeMainWindow();
              popToRoot({ clearSearchBar: true });
            }}`

**Comment by omarshahine** [comment]
@fturcheti I just tested with many files, jpgs etc and it worked just fine

**Comment by pernielsentikaer** [comment]
Hi @fturcheti 👋 

Could you give it another spin, should be fixed now

File: extensions/open-with-app/src/index.tsx [hunk 1]
```diff
@@ -47,10 +48,11 @@ function ApplicationsListItem(props: { application: Application }) {
             onAction={async () => {
               try {
                 const selectedItems = await getSelectedFinderItems();
+                await closeMainWindow({ clearRootSearch: true, popToRootType: PopToRootType.Suspended });
                 if (selectedItems.length != 0) {
-                  selectedItems.forEach((item) => {
-                    open(item.path.replace(/"/g, '\\"'), application.bundleId);
-                  });
+                  for (let i = 0; i < selectedItems.length; i++) {
+                    await open(selectedItems[i].path.replace(/"/g, '\\"'), application.bundleId);
```
Comment (fturcheti) [inline extensions/open-with-app/src/index.tsx:54]: We should remove these last leftovers from the old code. This call could be simplified to:

```js
await open(selectedItems[i].path, application);
```

**Comment by fturcheti** [comment]
Thanks @pernielsentikaer! I was `await`ing good news from you!

**Comment by pernielsentikaer** [comment]
Like this @fturcheti, or?

**Comment by fturcheti** [comment]
Yep. LGTM, @pernielsentikaer!

**Review by pernielsentikaer — APPROVED** [review]
Let's go 🔥

**Pull request merged**